functions/transformations/schema_mutators.go

```diff
@@ -206,20 +206,23 @@ func NewDropKeepMutator(qs flux.OperationSpec) (*DropKeepMutator, error) {
 }
 
 func (m *DropKeepMutator) checkColumns(tableCols []flux.ColMeta) error {
-	if m.DropCols != nil {
-		for c := range m.DropCols {
-			if err := checkCol(c, tableCols); err != nil {
-				return errors.Wrap(err, "drop error")
-			}
-		}
-	}
-
+	// Check if there are columns to keep. If there are, make sure they
+	// are present in the table columns. If there are no keep columns,
+	// then we must be performing the `drop` operation. Note that drop
+	// columns will be present even if we are performing the `keep`
+	// operation. So we must check the keep columns first.
 	if m.KeepCols != nil {
 		for c := range m.KeepCols {
 			if err := checkCol(c, tableCols); err != nil {
 				return errors.Wrap(err, "keep error")
 			}
 		}
+	} else if m.DropCols != nil {
```

Comment:
Looking at the code that does the mutation here:

https://github.com/influxdata/flux/blob/master/functions/transformations/schema_mutators.go#L267-L271

I think the calls to `checkColumns()` and `keepToDropCols()` should be transposed.

It seems like we are performing the check on the set of `dropCols` for the previous table, and that's what's causing this issue.  Your fix works, but it's still weird that we are calling `checkCols` on the stale set of drop columns.

Reply:
After discussing with @aanthony1243 we decided that dropping a non-existent column should not be an error. So the `m.DropCols != nil` check has been removed and now we only check columns in the case of `keep`.